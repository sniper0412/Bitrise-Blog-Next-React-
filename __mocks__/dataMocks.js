export const mockAuthors = [
  {
    slug: 'author-1',
    firstName: 'Bender',
    lastName: 'Rodr√≠guez',
    profileImage: 'http://bender.img',
    title: 'Chief Bending Robot',
    bio: 'A fictional character who is one of the main characters in the animated television series Futurama'
  }
];

export const mockCategories = [
  {
    slug: 'category-1',
    name: 'category 1'
  },
  {
    slug: 'category-2',
    name: 'category 2'
  }
];

export const mockTags = [
  {
    slug: 'tag-1',
    name: 'tag 1'
  },
  {
    slug: 'tag-2',
    name: 'tag 2'
  }
];

export const mockPosts = [
  {
    author: mockAuthors[0],
    slug: 'posts-1',
    seoTitle: 'post 1',
    metaDescription: 'Post 1 description',
    featuredImage: 'http://post-1.img',
    title: 'Post 1',
    body: '<h1>Post 1 body</h1>',
    categories: [mockCategories[0], mockCategories[1]],
    tags: [mockTags[0], mockTags[1]],
    published: 'Wed Jan 02 2019 16:50:20 GMT+0100',
    summary: 'Post 1 summary'
  }
];
