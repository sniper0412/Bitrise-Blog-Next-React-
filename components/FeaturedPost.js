import React from 'react';
import SVG from 'react-inlinesvg';

import { ROUTE_PATHS } from '../config';
import { PostType } from '../types';
import { formatDate } from '../utils';

const FeaturedPost = ({ slug, title, featuredImage, categories, published, summary }) => (
  <div className="featured-article">
    <div className="article-container">
      <a href={`/post/${slug}`}>
        <div className="image-container" style={{ backgroundImage: `url(${featuredImage})` }} />
      </a>

      <div className="article">
        <div className="article-categories">
          <a href="/tags/featured">
            <p className="category tag">featured</p>
          </a>
          {categories.map(({ name, slug }) => (
            <a className="category" key={slug} href={`${ROUTE_PATHS.categories}/${slug}`}>
              {name}
            </a>
          ))}
        </div>

        <h2>
          <a href={`/posts/${slug}`}>{title}</a>
        </h2>
        <h3>Published on {formatDate(published)}</h3>
        <p>{summary}</p>

        <div className="default-button more">
          <a className="read-more" title="read more..." href={`${ROUTE_PATHS.posts}/${slug}`}>
            <div className="button-content-wrapper">
              <span>Read more</span>
              <SVG src="/static/svg/right_arrow.svg" />
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
);

FeaturedPost.propTypes = PostType;

export default FeaturedPost;
