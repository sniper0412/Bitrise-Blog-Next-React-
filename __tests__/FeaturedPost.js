import renderer from 'react-test-renderer';

import { mockPosts } from 'dataMocks';
import FeaturedPost from '../components/FeaturedPost';

describe('FeaturedPost component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<FeaturedPost {...mockPosts[0]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
