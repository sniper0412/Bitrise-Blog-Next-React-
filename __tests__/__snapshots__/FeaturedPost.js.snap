// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedPost component renders correctly 1`] = `
<div
  className="featured-article"
>
  <div
    className="article-container"
  >
    <a
      href="/posts/posts-1"
    >
      <div
        className="image-container"
        style={
          Object {
            "backgroundImage": "url(http://post-1.img)",
          }
        }
      />
    </a>
    <div
      className="article"
    >
      <div
        className="article-categories"
      >
        <a
          href="/tags/featured"
        >
          <p
            className="category tag"
          >
            featured
          </p>
        </a>
        <a
          className="category"
          href="/categories/category-1"
        >
          category 1
        </a>
        <a
          className="category"
          href="/categories/category-2"
        >
          category 2
        </a>
      </div>
      <h2>
        <a
          href="/posts/posts-1"
        >
          Post 1
        </a>
      </h2>
      <h3>
        Published on 
        January 2, 2019
      </h3>
      <p>
        Post 1 summary
      </p>
      <div
        className="default-button more"
      >
        <a
          className="read-more"
          href="/posts/posts-1"
          title="read more..."
        >
          <div
            className="button-content-wrapper"
          >
            <span>
              Read more
            </span>
            <span
              className="isvg loading"
            />
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
